internal/engine/planner/planner.go

```diff
@@ -79,6 +81,43 @@ func CombinePlans(principalPolicyPlan, resourcePolicyPlan *PolicyPlanResult) *Po
 	}
 }
 
+func mergePlans(acc, current *PolicyPlanResult) *PolicyPlanResult {
+	if acc == nil {
+		return current
+	}
+	scopePermissions := current.ScopePermissions
+	allowFilter := current.AllowFilter
+	if current.AllowEmpty() {
+		scopePermissions = acc.ScopePermissions
+		allowFilter = acc.AllowFilter
+	} else if !acc.AllowEmpty() {
+		n := len(acc.AllowFilter) * len(current.AllowFilter)
+		allowFilter = make([]*qpN, 0, n)
+		for _, a := range acc.AllowFilter {
+			for _, c := range current.AllowFilter {
+				allowFilter = append(allowFilter, mkNodeFromLO(mkAndLogicalOperation([]*qpN{a, c})))
+			}
+		}
+	}
+	return &PolicyPlanResult{
+		Scope:            current.Scope,
+		ScopePermissions: scopePermissions,
+		AllowFilter:      allowFilter,
+		DenyFilter:       append(acc.DenyFilter, current.DenyFilter...),
+	}
+}
+
+func NewPolicyPlanResult(scope string, scopePermissions policyv1.ScopePermissions) *PolicyPlanResult {
```

Comment:
Could we pass the default scope permissions setting as an argument here? I'm a little worried that if this function was ever called from the context of a role policy evaluation, it would be set to the wrong default (the default for role policies is `SCOPE_PERMISSIONS_REQUIRE_PARENTAL_CONSENT_FOR_ALLOWS`).

Reply:
Good point. I think we can set the correct default permission in the caller function.